flink-python/pyflink/datastream/checkpoint_storage.py

```diff
@@ -0,0 +1,344 @@
+    (when high-availability is configured).
+
+        **Configuration**
+
+    As for all checkpoint storage, this type can either be configured within the application (by
+    creating the storage with the respective constructor parameters and setting it on the execution
+    environment) or by specifying it in the Flink configuration.
+
+    If the storage was specified in the application, it may pick up additional configuration
+    parameters from the Flink configuration. For example, if the backend if configured in the
+    application without a default savepoint directory, it will pick up a default savepoint
+    directory specified in the Flink configuration of the running job/cluster. That behavior is
+    implemented via the :func:`configure` method.
+    """
+
+    # The default maximal size that the snapshotted memory state may have (5 MiBytes).
+    DEFAULT_MAX_STATE_SIZE = 5 * 1024 * 1024
+
+    def __init__(self,
+                 checkpoint_path=None,
+                 max_state_size=None,
+                 j_jobmanager_checkpoint_storage=None):
+        """
+        Creates a new JobManagerCheckpointStorage, setting optionally the paths to persist
+        checkpoint metadata to, as well as configuring state thresholds.
+
+        WARNING: Increasing the size of this value beyond the default value
+        (:data:`DEFAULT_MAX_STATE_SIZE`) should be done with care.
+        The checkpointed state needs to be send to the JobManager via limited size RPC messages,
+        and there and the JobManager needs to be able to hold all aggregated state in its memory.
+
+        Example:
+        ::
+            >>> checkpoint_storage = JobManagerCheckpointStorage()
+
+        :param checkpoint_path: The path to write checkpoint metadata to. If none, the value from
+                                the runtime configuration will be used.
+        :param max_state_size: The maximal size of the serialized state. If none, the
+                               :data:`DEFAULT_MAX_STATE_SIZE` will be used.
+        :param j_jobmanager_checkpoint_storage: For internal use, please keep none.
+        """
+        if j_jobmanager_checkpoint_storage is None:
+            gateway = get_gateway()
+            JJobManagerCheckpointStorage = gateway.jvm.org.apache.flink.runtime.state.storage\
+                .JobManagerCheckpointStorage
+            JPath = gateway.jvm.org.apache.flink.core.fs.Path
+
+            if checkpoint_path is not None:
+                checkpoint_path = JPath(checkpoint_path)
+            if max_state_size is None:
+                max_state_size = JJobManagerCheckpointStorage.DEFAULT_MAX_STATE_SIZE
+            j_jobmanager_checkpoint_storage = JobManagerCheckpointStorage(checkpoint_path,
+                                                                          max_state_size)
+
+        self._j_jobmanager_checkpoint_storage = j_jobmanager_checkpoint_storage
```

Comment:
the instance variable `_j_jobmanager_checkpoint_storage ` could be removed and we could use the `_j_checkpoint_storage` defined in the super class CheckpointStorage